src/language-js/index.js

```diff
@@ -6,64 +6,70 @@ const options = require("./options");
 const createLanguage = require("../utils/create-language");
 
 const languages = [
-  createLanguage(require("linguist-languages/data/javascript"), {
+  createLanguage(require("linguist-languages/data/JavaScript"), {
     override: {
       since: "0.0.0",
       parsers: ["babel", "flow"],
-      vscodeLanguageIds: ["javascript"]
-    },
-    extend: {
-      interpreters: ["nodejs"]
+      vscodeLanguageIds: ["javascript"],
+      interpreters: ["node", "nodejs"]
     }
   }),
-  createLanguage(require("linguist-languages/data/javascript"), {
+  createLanguage(require("linguist-languages/data/JavaScript"), {
     override: {
       name: "Flow",
       since: "0.0.0",
       parsers: ["babel", "flow"],
       vscodeLanguageIds: ["javascript"],
-
       aliases: [],
       filenames: [],
-      extensions: [".js.flow"]
+      extensions: [".js.flow"],
+      interpreters: ["node"]
     }
   }),
-  createLanguage(require("linguist-languages/data/jsx"), {
+  createLanguage(require("linguist-languages/data/JSX"), {
     override: {
       since: "0.0.0",
       parsers: ["babel", "flow"],
       vscodeLanguageIds: ["javascriptreact"]
     }
   }),
-  createLanguage(require("linguist-languages/data/typescript"), {
+  createLanguage(require("linguist-languages/data/TypeScript"), {
     override: {
       since: "1.4.0",
       parsers: ["typescript"],
-      vscodeLanguageIds: ["typescript", "typescriptreact"]
+      vscodeLanguageIds: ["typescript"]
     }
   }),
-  createLanguage(require("linguist-languages/data/json"), {
+  createLanguage(require("linguist-languages/data/TSX"), {
+    override: {
+      since: "1.4.0",
+      parsers: ["typescript"],
+      vscodeLanguageIds: ["typescriptreact"]
+    }
+  }),
+  createLanguage(require("linguist-languages/data/JSON"), {
     override: {
       name: "JSON.stringify",
       since: "1.13.0",
       parsers: ["json-stringify"],
       vscodeLanguageIds: ["json"],
-
+      group: "JavaScript",
       extensions: [], // .json file defaults to json instead of json-stringify
       filenames: ["package.json", "package-lock.json", "composer.json"]
     }
   }),
-  createLanguage(require("linguist-languages/data/json"), {
+  createLanguage(require("linguist-languages/data/JSON"), {
     override: {
       since: "1.5.0",
       parsers: ["json"],
-      vscodeLanguageIds: ["json"]
+      vscodeLanguageIds: ["json"],
+      group: "JavaScript"
```

Comment:
`group: "JavaScript"` is unnecessary now

Reply:
removed